br/pkg/restore/split/client.go

```diff
@@ -42,61 +43,52 @@ const (
-		ScatterRegions(ctx context.Context, regionInfo []*RegionInfo) error
-		// GetOperator gets the status of operator of the specified region.
-		GetOperator(ctx context.Context, regionID uint64) (*pdpb.GetOperatorResponse, error)
-		// ScanRegions gets a list of regions, starts from the region that contains key.
-		// Limit limits the maximum number of regions returned.
-		ScanRegions(ctx context.Context, key, endKey []byte, limit int) ([]*RegionInfo, error)
-		// GetPlacementRule loads a placement rule from PD.
-		GetPlacementRule(ctx context.Context, groupID, ruleID string) (*pdhttp.Rule, error)
-		// SetPlacementRule insert or update a placement rule to PD.
-		SetPlacementRule(ctx context.Context, rule *pdhttp.Rule) error
-		// DeletePlacementRule removes a placement rule from PD.
-		DeletePlacementRule(ctx context.Context, groupID, ruleID string) error
-		// SetStoresLabel add or update specified label of stores. If labelValue
-		// is empty, it clears the label.
-		SetStoresLabel(ctx context.Context, stores []uint64, labelKey, labelValue string) error
-		// WaitRegionsScattered waits for an already started scatter region action to
-		// finish. Internally it will backoff and retry at the maximum internal of 2
-		// seconds. If the scatter makes progress during the retry, it will not decrease
-		// the retry counter. If there's always no progress, it will retry for about 1h.
-		// Caller can set the context timeout to control the max waiting time.
-		//
-		// The first return value is always the number of regions that are not finished
-		// scattering no matter what the error is.
-		WaitRegionsScattered(ctx context.Context, regionInfos []*RegionInfo) (notFinished int, err error)
-	}
-)
+// SplitClient is an external client used by RegionSplitter.
+type SplitClient interface {
+	// GetStore gets a store by a store id.
+	GetStore(ctx context.Context, storeID uint64) (*metapb.Store, error)
+	// GetRegion gets a region which includes a specified key.
+	GetRegion(ctx context.Context, key []byte) (*RegionInfo, error)
+	// GetRegionByID gets a region by a region id.
+	GetRegionByID(ctx context.Context, regionID uint64) (*RegionInfo, error)
+	// SplitWaitScatter splits a region from a batch of keys, waits for the split is
+	// finished, and scatters the new regions. It will return the
+	// original region, new regions and error. The input keys should not be encoded.
+	//
+	// The split step has a few retry times. If it meets error, the error is returned
+	// directly.
+	//
+	// The split waiting step has a backoff retry logic, if split has made progress,
+	// it will not increase the retry counter. Otherwise, it will retry for about 1h.
+	// If the retry is timeout, it will log a warning and continue.
+	//
+	// The scatter step has a few retry times. If it meets error, it will log a
+	// warning and continue.
+	SplitWaitScatter(ctx context.Context, region *RegionInfo, keys [][]byte) (*RegionInfo, []*RegionInfo, error)
```

Comment:
SplitAndWaitScatter?

Reply:
If we want to mention `And`, maybe `SplitAndWaitAndScatter` (I don't like the repeated `And`) or `SplitWaitAndScatter` (like in English "split, wait, and scatter"). Do you think `SplitWaitAndScatter` is better?

Reply:
lgtm